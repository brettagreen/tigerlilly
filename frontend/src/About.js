/**
 * 
 * @returns 
 */

function About() {

    return ("We're 'about' excellence in journalism. Any other questions?");
}

export default About;