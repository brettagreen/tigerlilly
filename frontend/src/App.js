import './css/App.css';
import TigerlillyApi from './api';
import { BrowserRouter } from 'react-router-dom';
import { useState } from 'react';
import UserContext from "./userContext";
import Nav from './Nav';
import TigerlillyRoutes from './TigerlillyRoutes';

function App() {

	function getLocalStorage() {
		const token = localStorage.getItem('userToken');
		let user = localStorage.getItem('user');

		if (user) user = JSON.parse(user); else user = null;
  
		TigerlillyApi.token = token
  
		return [token, user];
	}
  
	const [localToken, localUser] = getLocalStorage();
  
	const [userToken, setUserToken] = useState(localToken);
	const [user, setUser] = useState(localUser);

	function updateUserToken(value) {
		localStorage.setItem('userToken', value);
		TigerlillyApi.token = value;
		setUserToken(value);
	  }
	
	async function setCurrentUser(username) {
		if (!username) {
			localStorage.setItem('user', null);
			setUser(null);
		} else {
			const thisUser = await TigerlillyApi.getUser(username);
			console.log('thisUser.users', thisUser.users)
			localStorage.setItem('user', JSON.stringify(thisUser.users));
			setUser(thisUser.users);
		}
	
	}
	
	function profileUpdate(newUser) {
		setUser(newUser);
	}
	
	function isLoggedIn() {
		return user;
	}

	return (
		<UserContext.Provider value={{user, setCurrentUser}}>
			<div className="App">
				<BrowserRouter>
					<Nav isLoggedIn={isLoggedIn}></Nav>
					<TigerlillyRoutes profileUpdate={profileUpdate} isLoggedIn={isLoggedIn} updateUserToken={updateUserToken}/>
				</BrowserRouter>
			</div>
	  </UserContext.Provider>
	);
	
}

export default App;
