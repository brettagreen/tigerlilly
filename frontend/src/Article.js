import { useEffect, useState, useRef, useContext } from 'react';
//import { useParams, Link } from 'react-router-dom';
import { Link, TextField, ThemeProvider } from '@mui/material';
import UserContext from './userContext';
import TigerlillyApi from './api';
import { textareaTheme } from './css/styles';
import './css/article.css';

function Article({ passedArticle }) {

    const [article, setArticle] = useState(null);
    const [comments, setComments] = useState(null);
    const [keywords, setKeywords] = useState(null)
    const [showComments, setShowComments] = useState('show comments');
    const user = useContext(UserContext).user;

    const [commentForm, setCommentForm] = useState({userId: user ? user.id : 0, text: '', articleId: passedArticle.articleId});
    //const { id } = useParams();
    const showLeaveComment = useRef();
    const showCommentsRef = useRef();

    async function handleCommentSubmission(event) {
        event.preventDefault();
        showLeaveComment.current.hidden = true;
        console.log('user', user);
        console.log('commentForm', commentForm);
        const resp = await TigerlillyApi.commit('comments', commentForm, 'post');
        setComments([...comments].unshift(resp.comments));
        setCommentForm({...commentForm, 'text': ''});
    }

    function handleChange(event) {
        if (!isNaN(event.target.value) && event.target.value !== '') {
            setCommentForm({...commentForm, [event.target.name]: Number(event.target.value)});
        } else {
            setCommentForm({...commentForm, [event.target.name]: event.target.value});
        }
    }

    async function fetchComments(id) {
        console.log('fetchComments() useEffect')
        const resp = await TigerlillyApi.getComments(id, 'articles');

        if (resp.comments.length !== 0) {
            setComments(resp.comments);
        }
    }

    async function fetchKeywords(id) {
        console.log('fetchKeywords() useEffect');
        const resp = await TigerlillyApi.getArticleKeywords(id);

        setKeywords(resp.keywords);
    }

    function toggleComments() {
        if (showCommentsRef.current.hidden) {
            showCommentsRef.current.hidden = false;
            setShowComments('hide comments');
        } else {
            showCommentsRef.current.hidden = true;
            setShowComments('show comments');
        }
    }

    useEffect(() => {
        async function setUp() {
            setArticle(passedArticle);
            await fetchComments(passedArticle.articleId);
            await fetchKeywords(passedArticle.articleId);
        }
        setUp();
    }, [passedArticle]);

    return (
        <>
            {article ?
                <article>
                    <div>
                        <h2 id="articleTitle">{article.articleTitle}</h2>
                        {!article.authorHandle?<h4 style={{color: 'rgba(0, 0, 0, 0.87)'}}>by anonymous</h4>:<h4 style={{color: 'rgba(0, 0, 0, 0.87)'}}>by {<Link underline='none' href={`/author/${article.authorHandle}`}>
                            {article.authorFirst + ' ' + article.authorLast}</Link>}</h4>}
                        <p className="articleText">{article.text}</p>
                    </div>
                    <div>
                        <Link href="#" component="button" underline='always' onClick={() => { if (showLeaveComment.current.hidden) 
                                showLeaveComment.current.hidden = false 
                                else showLeaveComment.current.hidden = true}}>leave a comment
                        </Link>
                        &nbsp;&nbsp;
                        {comments !== null ?
                            <Link href="#" component="button" underline='always' onClick={toggleComments}>{showComments}</Link>
                        :null}
                    </div>
                </article>
            :null
            }
            <div hidden ref={showLeaveComment}>
                <form onSubmit={handleCommentSubmission}>
                    <ThemeProvider theme={textareaTheme}>
                        <TextField type="textarea" name="text" value={commentForm['text']}
                                    multiline minRows={5} onChange={handleChange} sx={{height: 'inherit'}} />
                    </ThemeProvider>
                        <br />
                        <Link href="#" sx={{float: 'left'}} onClick={() => showLeaveComment.current.hidden = true} 
                                type="button" component="button" underline="none">cancel</Link>
                        <Link href="#" sx={{float: 'right'}} component="button" type="submit" underline="none">submit</Link>
                </form>
            </div> 
            <br />
            {keywords ?
                <div>
                    {keywords.map((keyword, idx) => {
                        return(<Link key={-idx -1} sx={{display: "inline-block", fontSize: "x-small"}} 
                                    href={`/articleKeywords/${keyword['keyword']}`} underline="hover">
                                #{keyword['keyword']}&nbsp;
                            </Link>)
                        })
                    }
                </div>
            :null
            }
            <br />
            <div hidden ref={showCommentsRef}>
                {comments ? 
                    comments.map((comment, idx) => {
                        return(
                            <div key={idx+1}>
                                <h6>{<Link href={`/user/${comment['username']}`}>{comment['username']}</Link>}&nbsp;
                                    {new Date(comment['postDate']).toLocaleString()}
                                </h6>
                                <img src={`/icons/${comment['icon']}`} width={100} height={100} alt="user icon"/>
                                <p id="commentText">{comment['text']}</p>
                                <hr id="commentHR"></hr>
                            </div>
                        )
                    })
                :null   
                }
            </div>
        </>
    )
}

export default Article;