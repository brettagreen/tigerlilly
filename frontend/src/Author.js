import { useParams } from 'react-router-dom';
import { useEffect, useState } from 'react';
import TigerlillyApi from './api';
import './css/author.css';
import Link from '@mui/material/Link';

import { ListSubheader, List, ListItemButton, ListItemText, ListItem, Collapse } from '@mui/material';
import { ExpandLess, ExpandMore } from '@mui/icons-material'

function Author() {
    const { handle } = useParams();
    const [author, setAuthor] = useState(null);
    const [links, setLinks]  = useState(null);
    const [open, setOpen] = useState(false);

    useEffect(() => {
        async function fetchAuthor() {
            let res = await TigerlillyApi.getAuthor(handle);
            setAuthor(res.authors);
            res = await TigerlillyApi.getAuthorArticles(handle);
            setLinks(res.articles);
        }
        fetchAuthor();
    }, [handle]);
 
    return (<>
        {author?
            <>
            <h1 id="authorHeader">Hey! Let's give it up for the one and only {author.author}!</h1>
            <blockquote id="authorSlogan">
                "{author.authorSlogan}"
            </blockquote>
            <div style={{display: 'block', width: '100%'}}>
                <div style={{float: 'left', width: '30%'}}>
                    <List sx={{ float: 'left', width: '80%'}} aria-labelledby="nested-list-subheader">
                        <ListItemButton sx={{maxWidth: '120px'}} onClick={() => setOpen(!open)}>
                            <ListItemText primary="Author articles" />
                            {open ? <ExpandLess /> : <ExpandMore />}
                        </ListItemButton>
                        <Collapse in={open} timeout="auto" unmountOnExit>
                            {links ? links.map((link, idx) => {
                                return(
                                    <> 
                                        <ListItem key={idx}>
                                            <Link href={`/articles/${link.id}`} underline='hover' color='inherit'>{link.articleTitle}</Link>
                                            <Link href={`/issues/${link.issueId}`} underline='hover' color='inherit'>{link.issueTitle}</Link>
                                        </ListItem>
                                    </>
                                )
                            }) : null}
                        </Collapse>
                    </List>
                </div>
                <div style={{float: 'right', width: '70%'}}>
                    <div style={{float: 'right', margin: '10px'}}>
                        {author ? 
                            <div>
                                <img src={`/icons/${author.icon}`} width={250} height={250} alt="author icon"/>
                                <caption id="authorCaption">{`${author.author} aka ${author.authorHandle}`}</caption>
                            </div>
                        : null}
                    </div>
                    <div>
                        <p>
                            {author.authorBio}
                        </p>
                    </div>
                </div>
            </div>
            </>
        :null}</>
    );

}

export default Author;