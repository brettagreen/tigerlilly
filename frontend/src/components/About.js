/**
 *
 * @component /frontend/src/components/About
 * 
 * @description About component. Not sure what we'll have for an about page yet.
 * @author Brett A. Green <brettalangreen@proton.me>
 * @version 1.0
 * 
 * @returns {string} text. This is yet developed
 *
 */

function About() {

    return ("We're 'about' excellence in journalism. Any other questions?");
}

export default About;