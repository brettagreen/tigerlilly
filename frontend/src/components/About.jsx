/**
 *
 * @component /frontend/src/components/About
 * 
 * @description About component. Not sure what we'll have for an about page yet.
 * @author Brett A. Green <brettalangreen@proton.me>
 * @version 1.0
 * 
 * @returns {string} text. This is yet developed
 *
 */

function About() {

	return (<p style={{marginTop:'10vh'}}>"We're 'about' excellence in journalism. Any other questions?"</p>);
}

export default About;
