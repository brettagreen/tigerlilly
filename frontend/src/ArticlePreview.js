import { useEffect, useState } from 'react';
import { useParams } from 'react-router-dom';
import TigerlillyApi from './api';
import Grid from '@mui/material/Grid';
import Article from './Article';
import { gridTheme } from './css/styles';
import { ThemeProvider } from '@emotion/react';

function ArticlePreview() {
    const { keyword } = useParams();
    const [articles, setArticles] = useState(null);

    useEffect(() => {
        async function getArticles() {
            const res = await TigerlillyApi.getTaggedArticles(keyword);
            setArticles(res.articles);
        }
        getArticles();
    }, [keyword]);


    return (
        <>
            {articles ?
                <ThemeProvider theme={gridTheme}>
                    <h2>Articles matching #{keyword}</h2>
                    <Grid container rowSpacing={2} columnSpacing={{xs:2, sm:3, md:4}} columns={{xs:2, sm:3, md:4}}>

                        {articles.map((article, idx) => {
                            if ((article.text).split(' ').length >= 50) {
                                article.expand = true;
                            } else {
                                article.expand = false;
                            }
                            return ( 
                                <Grid key={idx} item xs={1} sm={1} md={1}>
                                    <Article passedArticle={article} issueArticle={false} />
                                </Grid>
                            )
                        })}
                    </Grid>
                </ThemeProvider>
            :null}
        </>
    )
}

export default ArticlePreview;