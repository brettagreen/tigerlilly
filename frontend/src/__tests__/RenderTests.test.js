import { render, fireEvent, screen, prettyDOM } from '@testing-library/react';
import { MemoryRouter, Route, Routes } from 'react-router-dom';
import About from '../components/About';
import Admin from '../components/Admin';
import App from '../components/App';
import Article from '../components/Article';
import ArticlePreview from '../components/ArticlePreview';
import Author from '../components/Author';
import ContactUs from '../components/ContactUs';
import Footer from '../components/Footer';
import React from 'react';

test('About', async function() {
    const { asFragment, findByText } = render(
        <About />
    );
    expect(await findByText("We're 'about' excellence in journalism. Any other questions?")).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
});

// test('renders Admin component', function() {
//   const {asFragment} = render(
//     <MemoryRouter initialEntries={[`/admin`]}>
//         <Routes>
//             <Route path="/admin" element={<Admin />} />
//         </Routes>
//     </MemoryRouter>
//   )
//   expect(asFragment()).toMatchSnapshot();
// });

test('App', async function() {
    const { findByAltText, findByRole, asFragment } = render(
        <App />
    );
    expect(await findByAltText('medium social')).toBeInTheDocument();
    const fbr = await findByRole('article');
    console.log('FIND BY ROLE', fbr);
    expect(fbr).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
});

test('Article, /articles/:id', async function() {
    const id = '1';
    const {findByText, findByTestId, asFragment} = render(
        <MemoryRouter initialEntries={[`/articles/${id}`]}>
            <Routes>
                <Route path="/articles/:id" element={<Article passedArticle={null} issueArticle={false} />} />
            </Routes>
        </MemoryRouter>
    );

    expect(await findByText('Jon Johnson')).toBeInTheDocument();
    expect(await findByTestId("testArticle")).toHaveTextContent('gobble gobble!');
    expect(asFragment()).toMatchSnapshot();
});

test('Article, w/ props and long text', async function() {
    const text = 'one two three four five six seven eight nine ten one two three four five six seven eight nine ten ' +
        'one two three four five six seven eight nine ten one two three four five six seven eight nine ten one two three four five ' +
        'six seven eight nine ten one two three four five six seven eight nine ten one two three four five six seven eight nine ten';
    const article = {articleId:1, articleTitle:'boy howdy what a day', text:text, authorFirst:'Bo',
                        authorLast:'Brand', authorHandle:'go fly a kite!'};
    article.expand = true;

    const {findByText, findByTestId, asFragment} = render(
        <Article passedArticle={article} issueArticle={false} />
    );

    expect(await findByTestId("preExpansion")).toHaveTextContent(
        'one two three four five six seven eight nine ten...read more'
    );

    fireEvent.click(await findByText('read more'));

    expect(await findByTestId("postExpansion")).toHaveTextContent(text);
    expect(asFragment()).toMatchSnapshot();

});

test('ArticlePreview', async function() {
    const keyword = 'funny';
    const {findByText, asFragment} = render(
        <MemoryRouter initialEntries={[`/articleKeywords/${keyword}`]}>
            <Routes>
                <Route path="/articleKeywords/:keyword" element={<ArticlePreview />} />
            </Routes>
        </MemoryRouter>
    );
    expect(await findByText("Articles matching #funny")).toBeInTheDocument();
    expect(await findByText("First test article")).toHaveClass("articleTitle");
    expect(asFragment()).toMatchSnapshot();

});

test('renders Author component', async function() {
    const handle = 'thejohnsonator';
    const {asFragment} = render(
        <MemoryRouter initialEntries={[`/author/${handle}`]}>
            <Routes>
                <Route path="/author/:handle" element={<Author />} />
                <Route path="/articles/:id" element={<Article />} />
            </Routes>
        </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
    
    console.log('BUTTON ROLE', await screen.findByRole('button'));
    // expect(await screen.findByText('issues:')).toBeInTheDocument();
    // fireEvent.click(await screen.findByTestId('issuesOpen'));
    // expect(await screen.findByText('First test article')).toBeInTheDocument();
    // let el = await screen.findByText('First test article');
    // console.log(prettyDOM(el));
    //fireEvent.click(el);
    //console.log(prettyDOM(document.body));
    //expect(await screen.findByTestId("paramsArticle")).toHaveTextContent('gobble gobble!');

});

test('renders ContactUs component', async function() {
    const {asFragment} = render(
        <MemoryRouter initialEntries={['/contact']}>
            <Routes>
                <Route path="/contact" element={<ContactUs />} />
            </Routes>
        </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
});

test('renders Footer component', function() {
    const {asFragment} = render(<Footer />);
    expect(asFragment()).toMatchSnapshot();
});
