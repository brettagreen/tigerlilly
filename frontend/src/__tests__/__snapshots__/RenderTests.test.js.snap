// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About 1`] = `
<DocumentFragment>
  We're 'about' excellence in journalism. Any other questions?
</DocumentFragment>
`;

exports[`Article, w/ props and long text 1`] = `
<DocumentFragment>
  <article>
    <div>
      <h2
        class="articleTitle"
      >
        boy howdy what a day
      </h2>
      <h4
        style="color: rgba(0, 0, 0, 0.87);"
      >
        by 
        <a
          class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-1g40t6m-MuiTypography-root-MuiLink-root"
          href="/author/go fly a kite!"
        >
          Bo Brand
        </a>
      </h4>
      <p
        data-testid="postExpansion"
      >
        one two three four five six seven eight nine ten one two three four five six seven eight nine ten one two three four five six seven eight nine ten one two three four five six seven eight nine ten one two three four five six seven eight nine ten one two three four five six seven eight nine ten one two three four five six seven eight nine ten
      </p>
    </div>
  </article>
  <br />
  <div>
    <a
      class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineHover css-11od7q1-MuiTypography-root-MuiLink-root"
      href="/articleKeywords/excellent"
    >
      #excellent 
    </a>
    <a
      class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineHover css-11od7q1-MuiTypography-root-MuiLink-root"
      href="/articleKeywords/funny"
    >
      #funny 
    </a>
    <a
      class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineHover css-11od7q1-MuiTypography-root-MuiLink-root"
      href="/articleKeywords/groaner"
    >
      #groaner 
    </a>
  </div>
</DocumentFragment>
`;

exports[`ArticlePreview 1`] = `
<DocumentFragment>
  <div
    class="PageHeader"
  >
    <h3>
      Articles matching #funny
    </h3>
  </div>
  <div
    class="MuiGrid-root MuiGrid-container css-jvssa7-MuiGrid-root"
    id="daGrid"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1 MuiGrid-grid-sm-1 MuiGrid-grid-md-1 css-unzqlu-MuiGrid-root"
    >
      <article>
        <div>
          <h2
            class="articleTitle"
          >
            First test article
          </h2>
          <h4
            style="color: rgba(0, 0, 0, 0.87);"
          >
            by 
            <a
              class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-1g40t6m-MuiTypography-root-MuiLink-root"
              href="/author/thejohnsonator"
            >
              Jon Johnson
            </a>
          </h4>
          <p
            data-testid="testArticle"
          >
            gobble gobble!
          </p>
        </div>
      </article>
      <br />
      <div>
        <a
          class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineHover css-11od7q1-MuiTypography-root-MuiLink-root"
          href="/articleKeywords/excellent"
        >
          #excellent 
        </a>
        <a
          class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineHover css-11od7q1-MuiTypography-root-MuiLink-root"
          href="/articleKeywords/funny"
        >
          #funny 
        </a>
        <a
          class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineHover css-11od7q1-MuiTypography-root-MuiLink-root"
          href="/articleKeywords/groaner"
        >
          #groaner 
        </a>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1 MuiGrid-grid-sm-1 MuiGrid-grid-md-1 css-unzqlu-MuiGrid-root"
    >
      <br />
    </div>
  </div>
</DocumentFragment>
`;

exports[`renders Author component 1`] = `<DocumentFragment />`;

exports[`renders ContactUs component 1`] = `
<DocumentFragment>
  <div
    class="PageHeader"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
      data-testid="FeedbackIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M20 2H4c-1.1 0-1.99.9-1.99 2L2 22l4-4h14c1.1 0 2-.9 2-2V4c0-1.1-.9-2-2-2zm-7 12h-2v-2h2v2zm0-4h-2V6h2v4z"
      />
    </svg>
    <h3>
      The Tigerlilly Online loooooves feedback!
    </h3>
    <h3>
      mad sad glad bad and allll those funky little vibes in between!
    </h3>
    <h3>
      Hey, you may (n)eve(n|r) hear back from us!
    </h3>
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
      data-testid="FeedbackIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M20 2H4c-1.1 0-1.99.9-1.99 2L2 22l4-4h14c1.1 0 2-.9 2-2V4c0-1.1-.9-2-2-2zm-7 12h-2v-2h2v2zm0-4h-2V6h2v4z"
      />
    </svg>
  </div>
  <div
    class="BackdropWrapper"
    style="min-width: 66vw; min-height: 50vh;"
  >
    <form
      autocomplete="off"
      enctype="multipart/form-data"
      novalidate=""
      style="margin: 1em;"
    >
      <div
        class="MuiFormControl-root MuiFormControl-marginNormal css-1iledgx-MuiFormControl-root"
      >
        <div
          class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-1qa8xn1-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="false"
            for=":r0:"
            id=":r0:-label"
          >
            name
            <span
              aria-hidden="true"
              class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
            >
               *
            </span>
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-1t7dcjg-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input css-1lkr5rz-MuiInputBase-input-MuiOutlinedInput-input"
              id=":r0:"
              name="name"
              required=""
              type="text"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-yjsfm1"
              >
                <span>
                  name *
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <div
          class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-1qa8xn1-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="false"
            for=":r1:"
            id=":r1:-label"
          >
            email
            <span
              aria-hidden="true"
              class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
            >
               *
            </span>
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-1t7dcjg-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input css-1lkr5rz-MuiInputBase-input-MuiOutlinedInput-input"
              id=":r1:"
              name="email"
              required=""
              type="email"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-yjsfm1"
              >
                <span>
                  email *
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <div
          class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="false"
            for=":r2:"
            id=":r2:-label"
          >
            what's on your mind?
            <span
              aria-hidden="true"
              class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
            >
               *
            </span>
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-multiline css-mj6wqb-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <textarea
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline css-1i66myl-MuiInputBase-input-MuiOutlinedInput-input"
              id=":r2:"
              name="feedback"
              required=""
              style="height: 0px; overflow: hidden;"
            />
            <textarea
              aria-hidden="true"
              class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline css-1i66myl-MuiInputBase-input-MuiOutlinedInput-input"
              readonly=""
              style="visibility: hidden; position: absolute; overflow: hidden; height: 0px; top: 0px; left: 0px; transform: translateZ(0); padding-top: 0px; padding-bottom: 0px; width: 100%;"
              tabindex="-1"
            />
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-yjsfm1"
              >
                <span>
                  what's on your mind? *
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <p
          class="MuiFormHelperText-root MuiFormHelperText-sizeMedium MuiFormHelperText-contained HiddenField css-4yhek5-MuiFormHelperText-root"
        >
          1000 characters left
        </p>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-1u8y31a-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="submit"
        >
          Submit
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </form>
  </div>
</DocumentFragment>
`;

exports[`renders Footer component 1`] = `
<DocumentFragment>
  <footer
    id="footer"
    style="display: block; margin-top: auto;"
  >
    <div
      style="float: left; margin-right: 4em;"
    >
      <img
        alt="TLO logo"
        src="/images/TLO_footer.jpg"
        width="100"
      />
    </div>
    <div
      style="float: left; position: relative; bottom: -7em;"
    >
      <div
        class="MuiGrid-root MuiGrid-container css-19sicoy-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-md-2 MuiGrid-grid-lg-2 MuiGrid-grid-xl-2 css-1b4xmg0-MuiGrid-root"
        >
          <a
            class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1p29rwh-MuiTypography-root-MuiLink-root"
            href="/about"
          >
            About
          </a>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-md-2 MuiGrid-grid-lg-2 MuiGrid-grid-xl-2 css-1b4xmg0-MuiGrid-root"
        >
          <img
            alt="tik tok social"
            src="/images/tiktok_icon.png"
            width="20"
          />
           TikTok
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-md-2 MuiGrid-grid-lg-2 MuiGrid-grid-xl-2 css-1b4xmg0-MuiGrid-root"
        >
          <a
            class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1p29rwh-MuiTypography-root-MuiLink-root"
            href="/contact"
          >
            Contact Us
          </a>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-md-2 MuiGrid-grid-lg-2 MuiGrid-grid-xl-2 css-1b4xmg0-MuiGrid-root"
        >
          <img
            alt="medium social"
            src="/images/medium_icon.png"
            width="20"
          />
           Medium.com
        </div>
      </div>
    </div>
    <div
      style="float: left; position: relative; bottom: -10.4em;"
    >
      Copyright 2024 The Tigerlilly Online
    </div>
  </footer>
</DocumentFragment>
`;
